{"ast":null,"code":"var _jsxFileName = \"/Users/marina/GitHub/calorie/src/Boundary.js\";\nimport React from 'react';\n\nclass Boundary extends React.Component {\n  constructor(props) {\n    //這邊不用'props'當參數就跑不出來？？\n    super(props);\n    this.state = {\n      hasError: false\n    };\n  } // 專門用來接error的life cycle，裡面不用再用try catch\n\n\n  componentDidCatch(error, errorInfo) {\n    console.log('1'); //根本無法console?\n\n    this.setState({\n      hasError: true\n    });\n    logErrorToMyService(error, errorInfo, 'test');\n    console.log('2');\n  }\n\n  render() {\n    if (this.state.hasError === true) {\n      return React.createElement(\"h1\", {\n        __source: {\n          fileName: _jsxFileName,\n          lineNumber: 22\n        },\n        __self: this\n      }, \"sth goes wrong.\");\n    }\n\n    return this.props.children;\n  }\n\n}\n\nexport default Boundary;","map":{"version":3,"sources":["/Users/marina/GitHub/calorie/src/Boundary.js"],"names":["React","Boundary","Component","constructor","props","state","hasError","componentDidCatch","error","errorInfo","console","log","setState","logErrorToMyService","render","children"],"mappings":";AAAA,OAAOA,KAAP,MAAkB,OAAlB;;AAEA,MAAMC,QAAN,SAAuBD,KAAK,CAACE,SAA7B,CAAuC;AACnCC,EAAAA,WAAW,CAACC,KAAD,EAAO;AAAI;AAClB,UAAMA,KAAN;AACA,SAAKC,KAAL,GAAa;AACTC,MAAAA,QAAQ,EAAG;AADF,KAAb;AAGH,GANkC,CAQnC;;;AACAC,EAAAA,iBAAiB,CAACC,KAAD,EAAQC,SAAR,EAAkB;AAC/BC,IAAAA,OAAO,CAACC,GAAR,CAAY,GAAZ,EAD+B,CACb;;AAClB,SAAKC,QAAL,CAAc;AAACN,MAAAA,QAAQ,EAAE;AAAX,KAAd;AACAO,IAAAA,mBAAmB,CAACL,KAAD,EAAQC,SAAR,EAAkB,MAAlB,CAAnB;AACAC,IAAAA,OAAO,CAACC,GAAR,CAAY,GAAZ;AACH;;AAEDG,EAAAA,MAAM,GAAE;AACJ,QAAI,KAAKT,KAAL,CAAWC,QAAX,KAAwB,IAA5B,EAAiC;AAC7B,aACI;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA,2BADJ;AAGH;;AACD,WAAO,KAAKF,KAAL,CAAWW,QAAlB;AAEH;;AAxBkC;;AA4BvC,eAAed,QAAf","sourcesContent":["import React from 'react';\n\nclass Boundary extends React.Component {\n    constructor(props){   //這邊不用'props'當參數就跑不出來？？\n        super(props);\n        this.state = {\n            hasError:  false,\n        }\n    }\n\n    // 專門用來接error的life cycle，裡面不用再用try catch\n    componentDidCatch(error, errorInfo){\n        console.log('1'); //根本無法console?\n        this.setState({hasError: true});\n        logErrorToMyService(error, errorInfo,'test')\n        console.log('2')\n    }\n\n    render(){\n        if (this.state.hasError === true){\n            return(\n                <h1>sth goes wrong.</h1>\n            )\n        }\n        return this.props.children\n        \n    }\n}\n\n\nexport default Boundary;"]},"metadata":{},"sourceType":"module"}